subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportsContainer.java

```diff
@@ -27,7 +27,7 @@
  * The reporting configuration for the {@link JacocoReport} task.
  */
 @Incubating
-public interface JacocoReportsContainer extends ReportContainer<Report> {
+public interface JacocoReportsContainer extends ReportContainer<ConfigurableReport> {
```

Comment:
I think this is actually a breaking change, as a number of signatures will change, eg `getAsMap()`.

You might instead use `withType(ConfigurableReport, Action<ConfigurableReport>)` instead of `all(Action<ConfigurableReport>)` when setting up the rules in the jacoco plugin.